### PR TITLE
✨ MQT Core

### PR DESCRIPTION
**Pull request by burgholzer**
This is the main PR that marks the transition from the existing submodule structure towards a unified `mqt-core` package and consists of a series of smaller PRs.

For more details on the individual changes please look at:

- [x] Transfer DD package from submodule to folder #316 
- [x] Transfer ZX package from submodule to folder #317 
- [x] Remove `mqt.qfr` Python package and corresponding CI #319 
- [x] Consolidate `googletest` submodule #320
- [x] Use stricter `clang-format` config throughout #321
- [x] Move DD package code to main directory #326 
- [x] Move ZX package code to main directory #327
- [x] Replace `QFR` with `MQT Core` (ensure backwards compatibility via alias) #328 
- [x] Various CI improvements and speedups 
  - #322 
  - #323 
  - #324
  - #325
  - #329

### CODE REVIEW

**Comment by burgholzer** [comment]
@hillmich @pehamTom This PR should now be ready for review. It might be best to quickly go through all the individual PRs that contributed towards this and check whether you have anything that you are not happy about.
As stated in the main description, functionality-wise, everything is just moved as-is and there should be no noticeable changes to the outside world. With the exception of the mqt.qfr package being dropped.
I'd still like to get your opinion or at least your approval of this, since it is a pretty big step from having everything split up into multiple repositories.

Once this PR is merged:
 - I'll submit PRs to the DDPackage and the ZX repository to add a disclaimer that development of the relevant feature continues here.
 - I'll change the GitHub repository name of this repository to `mqt-core` and submit a PR that changes all old links to new ones
 - Notify all relevant people that development of the DD and ZX package should continue here.
 - Update our top-level projects
 - Update our webpage/flyer
 - Anything else?

**Review by hillmich — APPROVED** [review]
The PR looks good to me, thank you for the effort.

Two questions came up for me while reviewing:

1. Should we create a new repository for MQT Core and archive this one?
2. Should the files retain a header stating the license?

**Pull request merged**
